build/v1/types.go

```diff
@@ -1022,23 +1026,37 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// BuildImageChangeTriggerKey captures the required elements for identifying the ImageStreamTag referenced by the more
+// generic ObjectReference BuildTriggerPolicy.ImageChange.From.  It is used by ImageChangeTriggerStatus, where a
+// specific instance of ImageChangeTriggerStatus in maintained in BuildConfigStatus.ImageChangeTriggers for each entry
+// in the BuildConfigSpec.Triggers array where the BuildTriggerPolicy.ImageChange pointer is set to a non-nil value
+type BuildImageChangeTriggerKey struct {
+	// namespace is the namespace where the ImageStreamTag used for an ImageChangeTrigger is located
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
+
+	// name is the name of the ImageStreamTag used for an ImageChangeTrigger
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+}
+
 // ImageChangeTriggerStatus tracks the latest resolved status of the associated ImageChangeTrigger policy
 // specified in the BuildConfigSpec.Triggers struct.
 type ImageChangeTriggerStatus struct {
-	// lastTriggeredImageID is the sha/id of the imageref cited in the 'from' field the last time this BuildConfig was triggered.
+	// lastTriggeredImageID represents the current sha/id of the ImageStreamTag cited in the 'from' field of a corresponding
+	// ImageChangeTrigger, captured by the non-nil BuildTriggerPolicy.ImageChange pointer from a BuildConfigSpec.Triggers
+	// entry, the last time a Build for this BuildConfig was instantiated as a result of *ANY* of the ImageChangeTriggers
```

Comment:
This seems to suggest the value isn't current, but actually a snapshot of the imageID the last time a build was instantiated. If it is instantiation, please list the subresource to use.

Reply:
OK I'm attempting to do this in the next push as well.  Though I'm not super confident I'm correctly translating your request into the godoc.  We'll find out soon enough.